extension/src/time_series/pipeline/lambda/lambda_expr.pest

```diff
@@ -0,0 +1,47 @@
+calculation = _{ SOI ~ let_expr ~ EOI }
+let_expr = { ("let" ~ var ~ "=" ~ tuple ~ ";")* ~ tuple }
+tuple = { binops ~ ("," ~ binops)* }
+binops = { unary ~ (operation ~ unary)* }
+unary = _{ neg | not | term }
+neg = { "-" ~ unary }
```

Comment:
```suggestion
neg = { "-" ~ term }
```
I would think we don't want to support things like `-----$val`.  Same comment for `not` on the next line.  Unless we're planning on expanding our selection of unary ops in the future, in which case ignore this comment.

Reply:
Unfortunately, I think `not not $foo` is a valid, if silly, expression in SQL so I think it’d be more surprising if we didn’t support it. (In general, most programming language seem to accept things like `!!foo` regardless of whether there’s a good reason to so.) (Long-term, if we ever extend this language to support postgres’s custom operators we’ll need to be able to support arbitrary combinations of prefix operators, and I think it’s better to define the grammar such that this is supported rather than risk backing ourselves into a corner)